packages/components/src/Notification/ToastNotification/context/ToastNotificationContext.tsx

```diff
@@ -44,18 +44,23 @@ export const ToastNotificationProvider = ({
   const [notifications, setNotifications] = useState<ToastNotificationObj[]>([])
 
   const addToastNotification: ToastNotificationContextValue["addToastNotification"] =
-    notification => {
-      const uuid = uuidv4()
-      const notificationWithId = { id: uuid, ...notification }
-
-      const notificationExists = notifications.find(
-        ({ id }) => id === notification.id
-      )
-
-      if (!notificationExists) {
-        setNotifications(existing => [...existing, notificationWithId])
-      }
-    }
+    useCallback(
+      notification => {
+        const uuid = uuidv4()
+        const notificationWithId = { id: uuid, ...notification }
+
+        setNotifications(existing => {
```

Comment:
From our testing, `notifications` is always empty, but due to our retry logic this function is called very quickly whilst notifications hasn't been updated yet. Due to that, we moved the "checking" logic inside the callback for setNotifications- as at that point, `existing` will have the correct value

Reply:
Hey Aaron, could you add a test for this fix please? :)

Reply:
Thanks @HeartSquared ! I'm not sure where I can add a test for this- should it be an interactive story? I can't see a spec file for toast's anywhere either

Reply:
Storybook is fine 😄 It's a bit of an awkward one to try and test in a spec file (given this was mostly legacy).

As a side note: Are you able to provide some steps to replicate the problem inside our repo (so that we can be sure we've actually fixed it)?

Reply:
Thanks @HeartSquared I've added an interactive story for this